src/main/utils/scanUtils.ts

```diff
@@ -158,8 +159,62 @@ export class ScanUtils {
         return exec.execSync(command, { cwd: cwd, maxBuffer: ScanUtils.SPAWN_PROCESS_BUFFER_SIZE, env: env });
     }
 
-    public static async executeCmdAsync(command: string, cwd?: string, env?: NodeJS.ProcessEnv | undefined): Promise<any> {
-        return await util.promisify(exec.exec)(command, { cwd: cwd, maxBuffer: ScanUtils.SPAWN_PROCESS_BUFFER_SIZE, env: env });
+    /**
+     * Executes a command asynchronously and returns the output.
+     * @param command - The command to execute.
+     * @param checkCancel - A function to check if cancellation is requested.
+     * @param cwd - The current working directory for the command execution.
+     * @param env - Optional environment variables for the command execution.
+     * @returns Command output or rejects with an error.
+     */
+    public static async executeCmdAsync(
+        command: string,
+        checkCancel: () => void,
+        cwd?: string,
+        env?: NodeJS.ProcessEnv | undefined
+    ): Promise<string> {
+        return new Promise<string>((resolve, reject) => {
+            try {
+                const childProcess: exec.ChildProcess = exec.exec(
+                    command,
+                    { cwd: cwd, maxBuffer: ScanUtils.SPAWN_PROCESS_BUFFER_SIZE, env: env },
+                    (error: exec.ExecException | null, stdout: string, stderr: string) => {
+                        clearInterval(checkCancellationInterval);
+                        if (error) {
+                            reject(error);
+                        } else {
+                            stderr.trim() ? reject(new Error(stderr.trim())) : resolve(stdout.trim());
+                        }
+                    }
+                );
+
+                const checkCancellationInterval: NodeJS.Timer = setInterval(() => {
+                    ScanUtils.cancelProcess(childProcess, checkCancel, reject);
+                }, ScanUtils.CANCELLATION_CHECK_INTERVAL_MS);
+            } catch (error) {
+                reject(error);
+            }
+        });
+    }
+
+    /**
+     * Cancels the child process if cancellation is requested.
+     * @param childProcess - The child process to be cancelled.
+     * @param checkCancel - A function to check if cancellation is requested.
+     * @param reject - A function to reject the promise.
+     */
+    private static cancelProcess(childProcess: exec.ChildProcess, checkCancel: () => void, reject: (reason?: any) => void): void {
+        try {
+            checkCancel();
```

Comment:
You already kill the process. Why running checkCancel?
Moreover - I think that running checkCancel without killing the process may end with a Zombie process.

Reply:
Rename to `cancelIfRequested`